pkg/util/logging/lager_adapter.go

```diff
@@ -0,0 +1,83 @@
+package logging
+
+import (
+	"fmt"
+
+	"code.cloudfoundry.org/lager"
+	"go.uber.org/zap"
+)
+
+// LagerAdapter satisfies the lager.Logger interface with zap as the
+// implementation.
+type LagerAdapter struct {
+	originalLogger *zap.Logger
+}
+
+// NewLagerAdapter returns a new lager.Logger that uses zap underneath.
+func NewLagerAdapter(zapLogger *zap.Logger) *LagerAdapter {
+	return &LagerAdapter{
+		originalLogger: zapLogger,
+	}
+}
+
+// RegisterSink is never used after initialization, so it does nothing.
+func (l *LagerAdapter) RegisterSink(_ lager.Sink) {}
+
+// Session returns a new logger with a nested session.
+func (l *LagerAdapter) Session(task string, data ...lager.Data) lager.Logger {
+	tmpLogger := l.originalLogger.Named(task)
+
+	if data != nil {
+		tmpLogger = l.originalLogger.With(dataToFields(data)...)
+	}
+
+	return &LagerAdapter{
+		originalLogger: tmpLogger,
+	}
+}
```

Comment:
not sure if this is implemented correctly, and I mostly don't care.

Reply:
Session is more useful in tracing and knowing where the logs are coming from in bigger projects where a lot of component logs come through or when you have many requests and you correlate all the logs for one request.

Reply:
Saying that, lager does a special thing with session so you can track it all the way down to where the original logger was created. With the adapter we won't have that so can't use Session in that way anymore.